src/main/java/org/checkstyle/autofix/parser/CheckConfiguration.java

```diff
@@ -0,0 +1,74 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle-openrewrite-recipes: Automatically fix Checkstyle violations with OpenRewrite.
+// Copyright (C) 2025 The Checkstyle OpenRewrite Recipes Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.autofix.parser;
+
+import java.util.List;
+import java.util.Map;
+
+public final class CheckConfiguration {
+    private final String name;
+    private final Map<String, String> properties;
+    private final List<CheckConfiguration> children;
+    private CheckConfiguration parent;
+
+    public CheckConfiguration(String name,
+                              Map<String, String> properties, List<CheckConfiguration> children) {
+        this.name = name;
+        this.properties = properties;
+        this.children = children;
+
+        for (CheckConfiguration child : children) {
+            child.setParent(this);
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public List<CheckConfiguration> getChildren() {
+        return children;
+    }
+
+    public String getProperty(String childName) {
```

Comment:
let's add more getxxx methods

`getIntProperty()`
`getBooleanProperty()`

so that we will hide data type conversion in this class and will avoid having boilerplate code across the codebase 

what do you think?